src/pool.jl

```diff
@@ -231,27 +167,46 @@ function Base.delete!(pool::CategoricalPool{S}, levels...) where S
         if haskey(pool.invindex, levelS)
             ind = pool.invindex[levelS]
             delete!(pool.invindex, levelS)
-            splice!(pool.index, ind)
-            ord = splice!(pool.order, ind)
-            splice!(pool.levels, ord)
+            splice!(pool.levels, ind)
```

Comment:
As for line 164 - do we want to allow `delete!`? I understood the preference was to create a new pool rather than altering an old pool in place?

Reply:
Yeah good catch. Actually I had spotted this but forgot to remove it. Let's say that was to check the quality of reviews. ;-)